x-pack/auditbeat/tracing/int_aligned.go

```diff
@@ -16,7 +16,7 @@ import (
 var errBadSize = errors.New("bad size for integer")
 
 func copyInt(dst unsafe.Pointer, src unsafe.Pointer, len uint8) error {
-	copy((*(*[maxIntSizeBytes]byte)(src))[:len], (*(*[maxIntSizeBytes]byte)(src))[:len])
+	copy((*(*[maxIntSizeBytes]byte)(dst))[:len], (*(*[maxIntSizeBytes]byte)(src))[:len])
```

Comment:
This can be brought into the modern era and have error handling added if it is
```
func copyInt(dst unsafe.Pointer, src unsafe.Pointer, len uint8) error {
	switch len {
	case 1, 2, 4, 8:
		copy(unsafe.Slice((*byte)(dst), len), unsafe.Slice((*byte)(src), len))
		return nil
	default:
		return errBadSize
	}
}
```
(we can also remove the array punning below by changing the preamble in `readInt` to `asSlice := unsafe.Slice((*byte)(ptr), len)`.

Reply:
That's great, but can we do this in a second future step? I've found this just by reading the code and don't have a proper setup in place to test more invasive changes.

Reply:
Yeah, fair enough.